catalog/core/catalog-core-attributeregistry/src/main/java/ddf/catalog/data/impl/AttributeRegistryImpl.java

```diff
@@ -56,14 +61,50 @@ public boolean register(final AttributeDescriptor attributeDescriptor) {
     @Override
     public void deregister(final String name) {
         notNull(name, "The attribute name cannot be null.");
-
         attributeMap.remove(name);
     }
 
     @Override
     public Optional<AttributeDescriptor> lookup(final String name) {
         notNull(name, "The attribute name cannot be null.");
-
         return Optional.ofNullable(attributeMap.get(name));
     }
+
+    public void setBundleContext(BundleContext bundleContext) {
+        this.bundleContext = bundleContext;
+    }
+
+    public BundleContext getBundleContext() {
+        return bundleContext;
+    }
+
+    public void bind(ServiceReference ref) {
```

Comment:
Since this is purpose built you can use more meaningful names for `bind` & `unbind` like `registerMetacardType`.

Also since you have no need for the ServiceReference you should use the ServiceObject directly (change your argument to `MetacardType` which is the default type returned from a Reference List.
http://aries.apache.org/modules/blueprint.html#reference-lists